### DISCUSSION TITLE
posts/2023/06/return-giscus/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#26] [Announcements]
# posts/2023/06/return-giscus/

The popular commenting platform is back on this site, with all its previous content intact.

https://www.brycewray.com/posts/2023/06/return-giscus/?utm_campaign=RSS&utm_source=RSS&utm_medium=RSS

**Comment by davidsneighbour** [comment]
Ah nice. I can't count the instances where I said to myself "wait, what? he wants me to write an email? good luck with that :P"

**Comment by idarek** [comment]
My thumbs up from myself. Either I am "older" generation folk, comment box removes a bit of struggle for busy people :)

ps. I will be stealing your approach to use `<details>` `<summary>` to show/hide comments :)

**Reply by brycewray** [reply]
@idarek Please note that (and partly with the help of @davidsneighbour) I have found something that works better. Still uses `<details><summary>` but, unlike the earlier method, isn't bad HTML yet does work in Firefox as well as Chrome and Safari.

https://github.com/brycewray/hugo-site/blob/main/layouts/partials/comments-giscus.html

**Comment by JingyiWu-codespace** [comment]
hello brycewray, recently I build personal website by hugo, add giscus as comment system, but I still meet github.com reject connection related to CSP, I render the gisuc module in local work, but in website like xxx.io, it does not work. so can you give me some advice about the problem? looking forward your message.

**Reply by brycewray** [reply]
@JingyiWu-codespace I include `https://giscus.app` in the following parts of my CSP: 

- `font-src`
- `frame-src`
- `script-src` — This also requires `unsafe-inline` due to giscus.
- `style-src` — This also requires `unsafe-inline` due to giscus.

I hope that information helps you.

**Reply by JingyiWu-codespace** [reply]
Thanks yo, I tried this in the meta tag in the HTML document, and it works.
